openjdk.test.modularity/src/com.stf/net/adoptopenjdk/test/modularity/UpgradeModPathTest.java

```diff
@@ -0,0 +1,405 @@
+		
+		switch (variant) {
+			/********************************************
+			 * 1) The following tests pertain to overriding 
+			 * using an exploded module folder 
+			 * ******************************/
+			case ExpDirModUpgrade : 
+				/*1.1) Override an application module*/
+				test.doRunForegroundProcess("Override an application module", "OT11", 
+						ECHO_ON, ExpectedOutcome.cleanRun().within("1m"), 
+							test.createJavaProcessDefinition()
+							.addModuleAddReads("com.test=ALL-UNNAMED")
+						    .addModuleToModulepath("openjdk.test.modularity/bin/common-mods")
+						    .addModuleToModulepath("openjdk.test.modularity/bin/tests")
+						    .addRootModule("com.test")
+						    .addModuleToUpgradeModulepath("openjdk.test.modularity/bin/common-override/com.hello")
+							.addPrereqJarToClasspath(JavaProcessDefinition.JarId.JUNIT)
+						    .addPrereqJarToClasspath(JavaProcessDefinition.JarId.HAMCREST)
+							.runClass(org.junit.runner.JUnitCore.class)
+							.addArg("com.ibm.runtimes.test.TestOverrideUsingAppModule")
+						);
+				
+				/*1.2) Override an upgradable platform module*/
+				test.doRunForegroundProcess("Override an upgradable platform module", "OT12", 
+						ECHO_ON, ExpectedOutcome.cleanRun().within("1m"), 
+							test.createJavaProcessDefinition()
+							.addModuleAddReads("com.test=ALL-UNNAMED")
+						    .addModuleToModulepath("openjdk.test.modularity/bin/common-mods")
+						    .addModuleToModulepath("openjdk.test.modularity/bin/tests")
+						    .addRootModule("com.test")
+						    .addModuleToUpgradeModulepath("openjdk.test.modularity/bin/common-override/java.transaction")
+							.addPrereqJarToClasspath(JavaProcessDefinition.JarId.JUNIT)
+						    .addPrereqJarToClasspath(JavaProcessDefinition.JarId.HAMCREST)
+							.runClass(org.junit.runner.JUnitCore.class)
+							.addArg("com.ibm.runtimes.test.TestOverrideUsingUpgradablePlatformModule")
+						);
+				
+				/*1.3) (Negative test) Override a non-upgradable platform module */
+				StfProcess ot13 = test.doRunForegroundProcess("(Negative test) Override a non-upgradable platform "
+						+ "module", "OT13", 
+						ECHO_ON, ExpectedOutcome.exitValue(1).within("1m"), 
+							test.createJavaProcessDefinition()
+							.addModuleAddReads("com.test=ALL-UNNAMED")
+						    .addModuleToModulepath("openjdk.test.modularity/bin/common-mods")
+						    .addModuleToModulepath("openjdk.test.modularity/bin/tests")
+						    .addRootModule("com.test")
+						    .addModuleToUpgradeModulepath("openjdk.test.modularity/bin/common-override/java.logging")
+							.addPrereqJarToClasspath(JavaProcessDefinition.JarId.JUNIT)
+						    .addPrereqJarToClasspath(JavaProcessDefinition.JarId.HAMCREST)
+							.runClass(org.junit.runner.JUnitCore.class)
+							.addArg("com.ibm.runtimes.test.TestOverrideUsingNonUpgradablePlatformModule")
+						);
+				
+				/*Note: This behavior is slightly different in OpenJDK in that, 

```

Comment:
Do we need to take this into account when running this test?  is it not applicable to openj9?